core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,241 @@
+      Session session = connection.openSession();
+      session.execCommand(execute);
+      try (InputStream stdout = new StreamGobbler(session.getStdout());
+           BufferedReader outBuffer = new BufferedReader(new InputStreamReader(stdout, Charsets.UTF_8))) {
+        Integer exitCode = session.getExitStatus();
+        if (exitCode != null && exitCode != 0) {
+          throw new IOException(String.format("Error running command %s on hostname %s; exit code: %d",
+                                              execute, config.oracleServerHostname, exitCode));
+        }
+        String out = CharStreams.toString(outBuffer);
+        //SQLPLUS command errors are not fetched from session.getStderr().
+        //Errors and output received after executing the command in SQLPlus prompt are the one that
+        //are printed on the SQL prompt
+        if (out.contains("ERROR at line")) {
+          throw new IOException(String.format("Error executing sqlplus query %s on hostname %s; error message: %s",
+                                              config.queryToExecute, config.oracleServerHostname, out));
+        }
+        Path file = new Path(config.pathToWriteFinalOutput);
+        FileSystem fs = FileSystem.get(file.toUri(), new Configuration());
+        try (
+          FSDataOutputStream outStream = fs.create(file);
+          BufferedWriter br = new BufferedWriter(new OutputStreamWriter(outStream, "UTF-8"))) {
+          br.write(out.replaceAll("(?m)^[\\s&&[^\\n]]+|[\\s+&&[^\\n]]+$", "")); //Remove multiline trailing spaces
+        }
+      }
+    } finally {
+      connection.close();
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
+  }
+
+  private String validateQuery(String query) {
+    if (!query.trim().endsWith(";")) {
+      query += ';';
+    }
+    return query;
+  }
+
+  private void init() {
+    StringBuilder scriptContent = null;
+    String dbConnectionString = config.dbUsername + "/" + config.dbPassword + "@" + config.oracleSID;
+    String colSeparator = getColSeparator(config.format);
+    scriptContent = new StringBuilder();
+    scriptContent.append("set colsep " + "\"" + colSeparator + "\"" + "\n");
+    scriptContent.append("set linesize 10000" + "\n");
+    scriptContent.append("set newpage none" + "\n");
+    scriptContent.append("set wrap off" + "\n");
+    scriptContent.append("set pagesize 0" + "\n");
+    scriptContent.append("set heading off" + "\n");
+    scriptContent.append("spool on" + "\n");
+    scriptContent.append(validateQuery(config.queryToExecute) + "\n");
+    scriptContent.append("spool off" + "\n");
+    scriptContent.append("exit");
+    //Set of commands to be executed in a session
+    String setPath = "export ORACLE_HOME=" + config.oracleHome + "; export ORACLE_SID=" + config.oracleSID + ";";
+    String createTempScriptToBeUsedForSpool = "echo '" + scriptContent.toString() + "' > /tmp/tmpHydrator.sql;";
```

Comment:
what happens if the ``/tmp`` directory does not exist, or the user does not have permissions to create files there?